diff --git a/src/command-runner.service.js b/src/command-runner.service.js
index v3.11.1..v3.11.1
--- a/src/command-runner.service.js
+++ b/src/command-runner.service.js
@@ -124,6 +124,9 @@
-            const { flags, description, defaultValue = undefined, required = false, choices = [], name: optionName = '', env = undefined, } = option.meta;
+            const { choices = [], conflicts = [], defaultIsPreset = false, defaultValue = undefined, defaultValueDescription, description, env = undefined, flags, hidden = false, implies = {}, name: optionName = '', preset, required = false } = option.meta;
             const handler = option.discoveredMethod.handler.bind(command.instance);
             const commandOption = new commander_1.Option(flags, description)
-                .default(defaultValue)
-                .preset(defaultValue)
-                .makeOptionMandatory(required);
+                .default(defaultValue, defaultValueDescription)
+                .preset(defaultIsPreset ? defaultValue : preset)
+                .implies(implies)
+                .conflicts(conflicts)
+                .makeOptionMandatory(required)
+                .hideHelp(hidden);
diff --git a/src/command-factory.interface.d.ts b/src/command-factory.interface.d.ts
index v3.11.1..v3.11.1
--- a/src/command-factory.interface.d.ts
+++ b/src/command-factory.interface.d.ts
@@ -1,0 +1,1 @@
+import type { CommanderError } from 'commander';
@@ -12,1 +12,1 @@
-    errorHandler?: ErrorHandler;
+    errorHandler?: (err: CommanderError) => void;
diff --git a/src/command-runner.interface.d.ts b/src/command-runner.interface.d.ts
index v3.11.1..v3.11.1
--- a/src/command-runner.interface.d.ts
+++ b/src/command-runner.interface.d.ts
@@ -3,1 +3,1 @@
-import { Command, CommandOptions } from 'commander';
+import { Command, CommandOptions, Option, OptionValues } from 'commander';
@@ -26,9 +26,15 @@
 export interface OptionMetadata {
+    choices?: string[] | true;
+    conflicts?: string | string[];
+    defaultIsPreset?: boolean;
+    defaultValue?: Option['defaultValue'];
+    defaultValueDescription?: string;
+    description?: string;
+    env?: string;
     flags: string;
-    description?: string;
-    defaultValue?: string | boolean | number;
+    hidden?: boolean;
+    implies?: OptionValues;
+    name?: string;
+    preset?: unknown;
     required?: boolean;
-    name?: string;
-    choices?: string[] | true;
-    env?: string;
 }
