diff --git a/dist/index.d.ts b/dist/index.d.ts
index v9.5.0..v9.5.0 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -335,7 +335,7 @@
  * Pick a list of properties from an object
  * into a new object
  */
-declare const pick: <T extends Record<string, unknown>, TKeys extends keyof T>(obj: T, keys: TKeys[]) => Pick<T, TKeys>;
+declare const pick: <T extends Record<string, any>, TKeys extends keyof T>(obj: T, keys: TKeys[]) => Pick<T, TKeys>;
 /**
  * Omit a list of properties from an object
  * returning a new object with the properties
